overlord/snapstate/snapmgr.go

```diff
@@ -667,6 +667,16 @@ func (m *SnapManager) doMountSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	defer func() {
```

Comment:
This might be `defer os.Remove(...)` directly, but I actually think we shouldn't defer it, because that will remove this file also when the setup fails for some reason. Right now there's no big difference, but once we better support reusing downloads (hopefully soon) this might prematurely kill a huge file which took some trouble to put on the local disk. It's a similar argument to the fact we don't actually "undo" the download right now.

So this would instead look more like:

```
err := ... SetupSnap ...
if err != nil {
        return err
}

err := os.Remove(...)
if err != nil {
        ... log but don't fail ...
}

return nil
```